org.hl7.fhir.publisher.core/src/main/java/org/hl7/fhir/igtools/renderers/XmlXHtmlRenderer.java

```diff
@@ -371,8 +373,23 @@ private String getNSAbbreviation(String namespace) throws IOException {
       return "";
     
     XMLNamespace ns = findByNamespace(namespace);
-    if (ns == null)
-      throw new IOException("Namespace "+namespace+" is not defined");
+    if (ns == null) {
+      if (autoNamespaces) {
+        String abbrev = null;
```

Comment:
This could be encapsulated into a generateNamespace method.

Reply:
it could, but there's no need.